### ISSUE TITLE
lettrine-xetex-bidi.def outdated

### ISSUE THREAD

**Issue by dflipo** [kvafa/bidi#31]
## Brief outline of the issue
lettrine-xetex-bidi.def patches an old definition of the internal command \@lettrine of the lettrine package leading to issues with the current implementation.

Rather than updating the patched definition in lettrine-xetex-bidi.def, I suggest to include the RTL flag mechanism for the \parshape command in the original definition of  \@lettrine.  I have an experimental version of  lettrine.sty, appended here.  Please let me know
[lettrine.sty.gz](https://github.com/kvafa/bidi/files/12793352/lettrine.sty.gz)
 if something like that would enable to get rid of lettrine-xetex-bidi.def?

## Check/indicate
- [ ] Relevant to the `xepersian` package
- [ x] [The `bidi` package issue tracker](https://github.com/kvafa/bidi/issues) has been searched for similar issues?
- [ ] [The `xepersian` package issue tracker](https://github.com/kvafa/xepersian/issues) has been searched for similar issues?
- [ ] Links to <tex.stackexchange.com> discussion if appropriate
- [ ] Links to <qa.parsilatex.com> discussion if appropriate

## Minimal example showing the issue

```tex
% !TEX TS-program = XeLaTeX
% !TEX encoding = UTF-8 Unicode


\documentclass{article}            % or some other class

  % Any packages other than the bidi package must be loaded here
\usepackage{lettrine}

  % The bidi package must be loaded as the last package
\usepackage{bidi}

  % Any preamble code goes here
  
\begin{document}

 \lettrine{V}{oici} un exemple 

\end{document}
```

## Log and PDF files  
! Undefined control sequence.
<argument> \LettrineFont \L@initial

**Comment by dflipo** [comment]
I append my experimental version of lettrine.sty in gzip format.  Comments and suggestions welcome!

[lettrine.sty.gz](https://github.com/kvafa/bidi/files/12792881/lettrine.sty.gz)

**Comment by kvafa** [comment]
Thanks for your bug report. Sure, I will certainly look into this but I'm afraid that will not be anytime soon as I am occupied by other things currently.

**Comment by EliseyP** [comment]
May be this post is answer? 

[https://github.com/kvafa/bidi/issues/13#issuecomment-1915377726](https://github.com/kvafa/bidi/issues/13#issuecomment-1915377726)

This [lettrine-xetex-bidi.def](https://mega.nz/file/LH5FWKJI#AxClpKl9TCgDUGA9czBbuZ_WzberWLupuh_srUv9b0I) in `texmf-local/tex/xelatex/bidi` solve this problem for me.

**Comment by dflipo** [comment]
NO, this file is incompatible with the current version of lettrine.sty [2024/01/20 v.2.60]. Lettrine.sty is aware of bidi now.
Please consider replacing lettrine-xetex-bidi.def by the appended (zipped) version.
[lettrine-xetex-bidi.zip](https://github.com/kvafa/bidi/files/14710236/lettrine-xetex-bidi.zip)

Patching an *internal* command of lettrine.sty (which is liable to change) is *definitely* not the way to go.

**Comment by EliseyP** [comment]
Thank you!  
Yes, it was some kind of "dirty hacking" - I had to compile my task urgently.

**Comment by ghost** [comment]
I have fixed this in version `40.1` of the `bidi` package and uploaded it about a week ago. It should already be available on TeXLive via update.

If this addresses your issue, then please consider closing this issue.

Please also note that this repository is unmaintained. The `bidi` package has a [new repository](https://github.com/xepersian/bidi).

**Comment by dflipo** [comment]
I confirm the bug is fixed in version 40.1 of the bidi package. Thanks!
